src/abTestingInfra/Variant/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { assertNameIsValid } from '../utils';
+import ExperimentContext from '../ExperimentContext';
+import { VariantProps } from '../types';
+import { recordWin } from '../emitter';
+
+function Variant(props: VariantProps): JSX.Element {
+  const { name, children } = props;
+
+  assertNameIsValid(name);
+
+  if (typeof children === 'function') {
+    return (
+      <ExperimentContext.Consumer>
+        {(experimentName) => {
+          const winFunction = (): void => recordWin(experimentName, name);
+
+          return children(winFunction);
+        }}
```

Comment:
Generally speaking folks are moving away from render props to hooks — can we? https://kentcdodds.com/blog/react-hooks-whats-going-to-happen-to-render-props/

Reply:
Good candidate for pair programming session as well!